### PR TITLE
DB-11113 fix LCC NPE when using getPreserveLineEndings in SpliceFileVTI

### PR DESCRIPTION
**Pull request by martinrupp**

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2216/

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by arnaud-splice — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java [hunk 1]
```diff
@@ -7545,8 +7545,7 @@ public TransactionController getTransactionCompile() {
             assert !booting:"booting is expected to be false";
 
             LanguageConnectionContext lcc=getLCC();
-            return lcc.getTransactionCompile();
-
+            return lcc == null ? null : lcc.getTransactionCompile();
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:7548]: Is this necessary to fix the problem or is it just preventing a similar NPE in the future?

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 2]
```diff
@@ -1124,7 +1124,7 @@ public boolean canWriteCache(TransactionController xactMgr) throws StandardExcep
         DDLWatcher ddlWatcher=driver.ddlWatcher();
         if(xactMgr==null)
             xactMgr = getTransactionCompile();
-        return ddlWatcher.canWriteCache((TransactionManager)xactMgr);
+        return xactMgr != null && ddlWatcher.canWriteCache((TransactionManager)xactMgr);
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java:1127]: Is this necessary to fix the problem or is it just preventing a similar NPE in the future?

**Review by dgomezferro — CHANGES_REQUESTED** [review]
If that code tries to prevent a similar NPE I think we should remove it, since it would hide errors.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java (hunk 1 above)
Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:7548]: good point, will check

Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:7548]: actually it is needed, `pass in language connection context` is not sufficient. if we remove this, `HdfsImportIT.testPropertiesCacheNPE` fails. However i agree that this is not completely understood. I'm creating a new jira.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java (hunk 2 above)
Comment (martinrupp) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java:1127]: see above

**Review by dgomezferro — APPROVED** [review]
Thanks for the explanation @martinrupp , I'm ok with merging the current fix if we make https://splicemachine.atlassian.net/browse/DB-11117 a priority for next sprint

**Comment by martinrupp** [comment]
DB-11117 will look at this more thoroughly

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
@OlegMazurov found a better solution in https://splicemachine.atlassian.net/browse/DB-11117

**Pull request closed without merging**